src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -794,16 +797,22 @@ private boolean handleJmxNotificationForCassandra22(
 
       case ERROR:
       case ABORT:
-        LOG.warn("repair session failed for segment with id '{}' and repair number '{}'", segmentId, repairNumber);
+        LOG.warn(
+            "repair session failed for segment with id '{}' and repair number '{}'",
+            segmentId,
+            repairNumber);
         failOutsideSynchronizedBlock = true;
         break;
 
       case COMPLETE:
         // This gets called through the JMX proxy at the end
         // regardless of succeeded or failed sessions.
         LOG.debug(
-            "repair session finished for segment with id '{}' and repair number '{}'", segmentId, repairNumber);
+            "repair session finished for segment with id '{}' and repair number '{}'",
+            segmentId,
+            repairNumber);
         condition.signalAll();
+        jmxProxy.removeRepairStatusHandler(repairNumber);
```

Comment:
Is this actually done in all code paths that are supposed to stop the SegmentRunner? Like failure cases. It seems to me like the status handler has to be remove or else `this` will never be garbage collected (since the status handler _is_ `this`). You've probably thought about this, I'm just asking for a double-check.